File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {

        unordered_map<int,int>sumf;
        int sum=0;
        int ans=0;
        sumf[0]=1;

        for(int i=0;i<nums.size();i++)
        {
            sum+=nums[i];
            if(sumf.count(sum-k)) ans+=sumf[sum-k];

            sumf[sum]++;
        }
return ans;
        
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:


        Rsum=0
        count=0
        indexMap=defaultdict(int)
        indexMap[0]=1
    
        for i in range(len(nums)):
            Rsum+=nums[i]
            diff=Rsum-k
            if diff in indexMap:
                count+=indexMap[diff]
            indexMap[Rsum]+=1
        return count




        